File: Developer/AppFrameworks/UIKit/YJCollectionView/YJCollectionView/YJTestCollectionViewCell.m
//
//  YJTestCollectionViewCell.m
//  YJCollectionView
//
//  Created by 阳君 on 16/5/21.
//  Copyright © 2016年 YJ. All rights reserved.
//

#import "YJTestCollectionViewCell.h"

@implementation YJTestCollectionCellModel

@end


@implementation YJTestCollectionViewCell

- (void)reloadDataAsyncWithCellObject:(YJUICollectionCellObject *)cellObject delegate:(YJUICollectionViewDelegate *)delegate {
    [super reloadDataAsyncWithCellObject:cellObject delegate:delegate];
    YJTestCollectionCellModel *cellModel = cellObject.cellModel;
    self.label.text = cellModel.index;
}

@end

File: Demo/OC/AppFrameworks/UIKit/YJCollectionView/YJCollectionView/Cell/YJTestCollectionViewCell.swift
//
//  YJTestCollectionViewCell.swift
//  YJCollectionView
//
//  Created by 阳君 on 2019/7/1.
//  Copyright © 2019 YJ. All rights reserved.
//

import UIKit
import YJCocoa

@objcMembers
class YJTestCollectionCellModel: NSObject {
    /// 位置
    var index: String!
}

@objcMembers
class YJTestCollectionViewCell: YJUICollectionViewCell {
    
    lazy var label: UILabel = {
        let _label = UILabel(frame: self.bounds)
        _label.font = UIFont.systemFont(ofSize: 20)
        _label.textAlignment = .center
        return _label
    }()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor.green
        self.contentView.addSubview(self.label)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        self.label.frame = self.bounds
    }
    
    override func collectionViewManager(_ collectionViewManager: YJUICollectionViewManager, reloadWith cellObject: YJUICollectionCellObject) {
        super.collectionViewManager(collectionViewManager, reloadWith: cellObject)
        let cm = cellObject.cellModel as? YJTestCollectionCellModel
        self.label.text = cm?.index
    }
    
}
